src/main/java/com/minecolonies/coremod/client/gui/WindowBuildBuilding.java

```diff
@@ -0,0 +1,362 @@
+     * Contains all resources needed for a certain build.
+     */
+    private final HashMap<String, ItemStorage> resources = new HashMap<>();
+
+    /**
+     * Drop down list for style.
+     */
+    private       DropDownList     stylesDropDownList;
+
+    /**
+     * List of style for the section.
+     */
+    @NotNull
+    private List<String> styles = new ArrayList<>();
+
+    /**
+     * Constructor for the window when the player wants to hire a worker for a certain job.
+     *
+     * @param c          the colony view.
+     * @param buildingId the building position.
+     */
+    public WindowBuildBuilding(final ColonyView c, final BlockPos buildingId)
+    {
+        super(Constants.MOD_ID + BUILDING_NAME_RESOURCE_SUFFIX);
+        this.colony = c;
+        building = (AbstractBuildingView) colony.getBuilding(buildingId);
+        initStyleNavigation();
+        registerButton(BUTTON_BUILD, this::confirmClicked);
+        registerButton(BUTTON_CANCEL, this::cancelClicked);
+        final Button buttonBuild = findPaneOfTypeByID(BUTTON_BUILD, Button.class);
+        if (building.getBuildingLevel() == 0)
+        {
+            buttonBuild.setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.build"));
+        }
+        else
+        {
+            buttonBuild.setLabel(LanguageHandler.format("com.minecolonies.coremod.gui.workerHuts.upgrade"));
+        }
+    }
+
+    /**
+     * On cancel button.
+     */
+    private void cancelClicked()
+    {
+        building.openGui(false);
+    }
+
+    /**
+     * On confirm button.
+     */
+    private void confirmClicked()
+    {
+        MineColonies.getNetwork().sendToServer(new BuildingSetStyleMessage(building, styles.get(stylesDropDownList.getSelectedIndex())));
+        MineColonies.getNetwork().sendToServer(new BuildRequestMessage(building, BuildRequestMessage.BUILD));
+        cancelClicked();
+    }
+
+    /**
+     * Update the styles list but try to keep the same one.
+     */
+    private void updateStyles()
+    {
+        styles = Structures.getStylesFor(building.getSchematicName());
+        int newIndex = styles.indexOf(building.getStyle());
+        if (newIndex == -1)
+        {
+            newIndex = 0;
+        }
+
+        final boolean enabled;
+        if(Settings.instance.isStaticSchematicMode())
+        {
+            enabled = false;
+        }
+        else
+        {
+            enabled = styles.size() > 1;
+        }
+
+        findPaneOfTypeByID(BUTTON_PREVIOUS_STYLE_ID, Button.class).setEnabled(enabled);
+        findPaneOfTypeByID(DROPDOWN_STYLE_ID, DropDownList.class).setEnabled(enabled);
+        findPaneOfTypeByID(BUTTON_NEXT_STYLE_ID, Button.class).setEnabled(enabled);
+        stylesDropDownList.setSelectedIndex(newIndex);
+    }
+
+    /**
+     * Clears and resets/updates all resources.
+     */
+    private void updateResources()
+    {
+        final World world = Minecraft.getMinecraft().world;
+        resources.clear();
+
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 198 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)